workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/add_derived.py

```diff
@@ -71,7 +71,10 @@ def _column_pq2(ds: xr.Dataset) -> xr.DataArray:
 
 
 def _column_dq1(ds: xr.Dataset) -> xr.DataArray:
-    if "net_heating" in ds:
+    if "net_heating_due_to_machine_learning" in ds:
+        column_dq1 = ds.net_heating_due_to_machine_learning
+    elif "net_heating" in ds:
```

Comment:
This conditional logic is buried pretty deep in the call stack. If we need to maintain backwards compatibility, I would recommend renaming "net_heating" to "net_heating_due_to_machine_learning" somewhere more obvious (e.g. closer to where the data is loaded/queried).

Also, it would be nice to not maintain this backwards compatibility forever, so I would add some kind of deprecation warning, and/or making a github issue to delete it in a couple of months.

Reply:
I think this an appropriate place for this logic because there is a lot of other similar logic in this module. E.g. for getting the correct <nQx> variable from both nudge-to-fine or nudge-to-obs runs. For example:

https://github.com/VulcanClimateModeling/fv3net/blob/d65e0dc97d3f3696d65e352e52a7e57a008c0c8b/workflows/prognostic_run_diags/fv3net/diagnostics/prognostic_run/add_derived.py#L160-L169

I'll make a GitHub issue to remember to delete it in the future.